libtransmission/web.cc

```diff
@@ -332,6 +345,21 @@ class tr_web::Impl
         (void)curl_easy_setopt(e, CURLOPT_NOSIGNAL, 1L);
         (void)curl_easy_setopt(e, CURLOPT_PRIVATE, task);
 
+#if LIBCURL_VERSION_NUM <= CURL_VERSION_BITS(7, 76, 1)
+        /*
+         * Up to Curl 7.76.1, if we explicitly choose the IP version we want to
+         * use, it is still possible that the wrong IP version is used. To work
+         * around the issue, we must disable DNS cache and connection reuse.
+         */
+        if (task->ipProtocol() != CURL_IPRESOLVE_WHATEVER)
+        {
+            (void)curl_easy_setopt(e, CURLOPT_DNS_CACHE_TIMEOUT, 0L);
+            (void)curl_easy_setopt(e, CURLOPT_FORBID_REUSE, 1L);
+        }
+#endif
```

Comment:
This is concerning, since this version was released about 9 months ago and isn't on most Linux distros / on any macOS release yet, so for most people who want to use ipv6 it looks like nothing will get cached -- wouldn't this require a new DNS request for every announce?

https://github.com/transmission/transmission/issues/1815 is already unhappy with the current number of DNS requests so I can imagine what a new dns lookup per announce would be like :smiley_cat:

Reply:
[Pretty bad](https://github.com/transmission/transmission/issues/1815#issuecomment-1046344628). But it is in [bullseye-backports](https://packages.debian.org/bullseye-backports/libcurl4-openssl-dev) and also in [AUR](https://aur.archlinux.org/packages/libcurl-openssl-1.0-8) though so you could theoretically increase the minumum.

Reply:
Yes, the number of DNS requests would increase.

The only alternative I see is to disable the double request in announce-http.cc if libcurl version is wrong.

Reply:
Ubuntu already has an LTS version with a recent enough curl.